models/users.js

```diff
@@ -52,10 +52,15 @@ const addOrUpdate = async (userData, userId = null) => {
     }
 
     // userId is null, Add or Update user
-    const user = await userModel.where("github_id", "==", userData.github_id).limit(1).get();
-    if (!user.empty) {
+    let user;
+    if (userData.github_user_id) {
+      user = await userModel.where("github_user_id", "==", userData.github_user_id).limit(1).get();
+    }
+    if (!user || (user && user.empty)) {
+      user = await userModel.where("github_id", "==", userData.github_id).limit(1).get();
+    }
+    if (user && !user.empty) {
```

Comment:
It would be better if we did not put this behind the feature flag. As it is not going to affect the existing flow and will automatically add the `github_user_id` on every login.

Reply:
Also, [this helper function](https://github.com/Real-Dev-Squad/website-backend/blob/d4ae3ac75ef093650cf33321dc7041b98f7acf50/models/users.js#L33) has a vast dependency throughout the code. I am not sure how can I pass the `dev` parameter as it also has an optional parameter.